apps/frontend/src/components/technique/TechniqueTable.tsx

```diff
@@ -0,0 +1,234 @@
+import { FunctionType } from 'utils/utilTypes';
+
+import AssignedInstrumentsTable from './AssignedInstrumentsTable';
+import AssignInstrumentsToTechniques from './AssignInstrumentsToTechniques';
+import CreateUpdateTechnique from './CreateUpdateTechnique';
+import {
+  InstrumentFragment,
+  TechniqueFragment,
+  UserRole,
+} from '../../generated/sdk';
+
+const columns = [
+  {
+    title: 'Name',
+    field: 'name',
+  },
+  {
+    title: 'Short code',
+    field: 'shortCode',
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Instruments',
+    field: 'instruments.length',
+    emptyValue: '-',
+  },
+];
+
+const TechniqueTable = () => {
+  const {
+    loadingTechniques,
+    techniques,
+    setTechniquesWithLoading: setTechniques,
+  } = useTechniquesData();
+
+  const { api } = useDataApiWithFeedback();
+  const isUserOfficer = useCheckAccess([UserRole.USER_OFFICER]);
+  const [urlQueryParams, setUrlQueryParams] =
+    useQueryParams<UrlQueryParamsType>(DefaultQueryParams);
+
+  const [openTechniqueAssignment, setOpenTechniqueAssignment] = useState(false);
+  const [selectedTechniques, setSelectedTechniques] = useState<
+    TechniqueFragment[]
+  >([]);
+
+  const onTechniqueDelete = async (techniqueDeletedId: number | string) => {
+    try {
+      await api({
+        toastSuccessMessage: 'Technique deleted successfully!',
+      }).deleteTechnique({
+        id: techniqueDeletedId as number,
+      });
+
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  const AssignmentIndIcon = (): JSX.Element => <AssignmentInd />;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function setAssigningTechniqueId(technique: TechniqueFragment): void {
+    const techniqueList: TechniqueFragment[] = [];
+    techniqueList.push(technique);
+    setSelectedTechniques(techniqueList);
+    setOpenTechniqueAssignment(true);
+  }
+
+  const assignInstrumentsToTechniques = async (
+    instruments: InstrumentFragment[]
+  ): Promise<void> => {
+    const techniqueId = selectedTechniques[0].id;
+    if (instruments?.length) {
+      await api({
+        toastSuccessMessage: `Instrument/s assigned to the selected technique successfully!`,
+      }).assignInstrumentsToTechnique({
+        instrumentIds: instruments.map((instrument) => instrument.id),
+        techniqueId,
+      });
+    }
+    setTechniques((techniques) =>
+      techniques.map((techniqueItem) => {
+        if (techniqueItem.id === techniqueId) {
+          return {
+            ...techniqueItem,
+            instruments: [...techniqueItem.instruments, ...instruments],
+          };
+        } else {
+          return techniqueItem;
+        }
+      })
+    );
+  };
+
+  const AssignedInstruments = React.useCallback(
+    ({ rowData }) => {
+      return <AssignedInstrumentsTable technique={rowData} />;
+    },
+    [setTechniques]
+  );
+
+  const removeIntrumentsFromTechnique = async (
```

Comment:
Since this follows the same pattern as `assignInstrumentsToTechniques`. Kindly check the comments attach to the `assignInstrumentsToTechniques`.